ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/ConnectorList.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import com.google.errorprone.annotations.Immutable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Immutable
+public class ConnectorList extends KsqlEntity {
+
+  private final ImmutableMap<String, SimpleConnectorInfo> connectors;
+
+  @JsonCreator
+  public ConnectorList(
+      @JsonProperty("statementText")  final String statementText,
+      @JsonProperty("warnings")       final List<KsqlWarning> warnings,
+      @JsonProperty("connectors")     final Map<String, SimpleConnectorInfo> connectors
+  ) {
+    super(statementText, warnings);
+    this.connectors = ImmutableMap.copyOf(Objects.requireNonNull(connectors, "connectors"));
+  }
+
+  public ImmutableMap<String, SimpleConnectorInfo> getConnectors() {
```

Comment:
good catch - when I was originally coding this up I thought the value would be null for those that we couldn't get information for, but now that i've created SimpleConnectorInfo this can be simplified